src/components/ChatContent/Tabs/Tab.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <div style="height: 100%">
+    <div
+      ref="scrollDiv"
+      style="padding: 5.6px 7px;"
+      :style="{height: `calc(100% - ${footerHeight}px - 5.6px * 2)`}"
+    >
+      <VuePerfectScrollbar
+        class="scroll-area"
+        :settings="settings"
+        @ps-scroll-y="scrollHandle"
+      >
+        <div style="height: 100%; word-wrap: break-word; white-space: normal;">
+          <slot ref="slot" />
+        </div>
+      </VuePerfectScrollbar>
+    </div>
+
+    <div ref="footer" style="overflow: auto; padding: 5px; 10px;">
+      <slot
+        name="footer"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from '@/types/vue';
+  import Component from 'vue-class-component';
+  import VuePerfectScrollbar from 'vue-perfect-scrollbar';
+
+  @Component({
+    components: {
+      VuePerfectScrollbar,
+    },
+  })
+  export default class Tab extends Vue {
+    settings = {
+      maxScrollbarLength: 60,
+    };
+
+    $refs!: {
+      afterScrollbar: Vue;
+      footer: HTMLElement;
+    }
+
+    autoScroll = true;
+
+    container!: Element;
+
+    footerHeight: number = 0;
+
+    scrollHandle(evt: any) {
+      const { container } = this;
+      const actualScrollHeight = container!.scrollHeight - this.container.clientHeight;
+      if (evt.type === 'ps-scroll-y') {
+        this.$forceUpdate();
+        if (container!.scrollTop > actualScrollHeight - 50) {
+          this.autoScroll = true;
+        } else this.autoScroll = false;
+      }
+    }
+
+    onContentChanged() {
+      if (this.autoScroll) {
+        this.$nextTick(() => {
+          this.container!.scrollTop = this.container!.scrollHeight;
+        });
+      }
+    }
+
+    updateFooterHeight() {
+      this.footerHeight = this.$refs.footer.clientHeight;
+      this.onContentChanged();
+    }
+
+    created() {
+      this.$store.subscribe((mutation, state) => {
```

Comment:
Should the visibility be passed as a prop in addition to the v-show?
I don't think that I want to put the v-show inside of this file, but I also don't think that this component should be in charge of listening to tab switches, as the v-show is in ChatContent.

Reply:
Is there any reason why you can't _just_ use v-show in the outer file?

Reply:
Because a component can't listen to a v-show, the tab won't be able to know when it becomes visible without either a prop or listening to the event.

Reply:
Oh, because it needs to update its scroll. Maybe you should just go for v-if, that way you can use `mounted`. In the future, if we have like, 30 channels or something it'd probably be more performant anyways. Give it a test though to check if it feels responsive enough.

Reply:
The scrollbar's position isn't retained that way. I should probably save it in the store, but that will probably happen later on, unless you think that's urgent enough

Reply:
(For reference - I kept the v-show and added a `visibile` property)